paperqa/core.py

```diff
@@ -115,3 +115,50 @@ async def map_fxn_summary(
         ),
         llm_result,
     )
+
+async def gather_with_batch(
```

Comment:
Can we merge this with `gather_with_concurrency`? This is still concurrent gathering, just the inner workings are via batching.

Reply:
`gather_with_concurrency ` is a wrapper to actually use `map_fxn_summary`. I think we can unify `gather_with_batch` and  `map_fxn_summary` because they work very similarly. But because the inputs and returns are different, I kept them separated to make the the static type definition easier